erts/emulator/beam/erl_alloc_util.h

```diff
@@ -694,9 +698,11 @@ struct Allctr_t_ {
         ErtsAlcCPoolData_t  *sentinel;
 
 	UWord		abandon_limit;
+        UWord           discard_limit;
```

Comment:
`Allctr.cpool.discard_limit` do not seem to be used other than in ASSERT.

The `abandon_limit` check is done both on allocator level and on carrier, but the `discard_limit` check is only done on carrier.